### ISSUE TITLE
Chaining populate for two associations fails

### ISSUE THREAD

**Issue by cbarlow1993** [emahuni/sails-hook-deep-orm#3]
Chaining populations as in the example fails with error: ```TypeError: Image.findOne(...).populate(...).populate is not a function```

```
    var image = await Image.findOne({
        id: inputs.imageId
      })
      .populate('task.images')
      .populate('project.classes')

```
I believe anything that is chained after a deep population fails unless it is a .catch(). See related https://github.com/emahuni/sails-hook-deep-orm/issues/2

**Comment by emahuni** [comment]
Yes it should fail, it's not chainable since it settles the deffered object. This is similar to the `stream` method in Sails, there is no way to return the deffered obj 
 or promise once you use deep populations, that's why I made inclusive populations. It's only Polymorphic populations without deep populations that allow you to chain since the deffered obj is returned to you.

Do this instead:

```js
var image = await Image.findOne({
        id: inputs.imageId
      })
      .populate('task&project.images&classes')
```

What you are essentially saying here is that `task` and `project` are populated on `Image` records and `images` and `classes` are populated on `task` and `project` records.

The order doesn't matter, it'll try all permutations, ie: `task->images`, then `task->classes`, then `project->images` and finally `project->classes`.

The reason why this is happening is becoz I can't get model instances at that high level, plus the deffered obj issue is also because of the high level of implementation. If this was deep in Waterline this wasn't supposed to happen. I am sure you could chain them. So there are sacrifices done there. But it gives a very powerful query language.

I think I should clearly state this in the documentation.
